### ISSUE TITLE
Catalyst Support

### ISSUE THREAD

**Issue by JonasGessner** [beeware/Python-Apple-support#117]
The makefile doesn't seem to support Catalyst. It would be nice to have support for this target!

**Comment by freakboy3742** [comment]
Thanks for the suggestion - but I'm not even sure what this would mean. Python-Apple-support is a library that you would include *in* your iOS project... unless I'm missing something, it shouldn't matter whether your project is a Catalyst project or a   Cocoa project - you link in the Python support library, and invoke it as required.

What exactly is it that you think needs to be added?

**Comment by JonasGessner** [comment]
Yes but to be able to link the python library when building an app for catalyst the library itself needs to be built specifically for catalyst as well. iOS or macOS libraries can't be used in catalyst builds.

To build a library for catalyst you have to build against the macOS SDK and specify a target in the form `x86_64-apple-ios13.0-macabi`, with the arch and iOS version adjusted as needed.

**Comment by freakboy3742** [comment]
Ah - I didn't realise Catalyst was based on an entirely different ABI. 

Catalyst isn't a huge priority for me personally, but if you wanted to add another target platform to support Catalyst, I wouldn't object to a PR that added it.

**Comment by freakboy3742** [comment]
For anyone interested in adding MacCatalyst support, user @asavva-2016 was working on a series of patches to implement this feature:
* https://github.com/beeware/cpython-apple-source-deps/pull/43
* https://github.com/freakboy3742/cpython/pull/5
* https://github.com/beeware/Python-Apple-support/pull/256

These were getting close to being ready to merge, but @asavva-2016 wasn't able to commit the time to get them over the line.

**Comment by freakboy3742** [comment]
The first part of this work - adding macCatalyst binaries to cpython-apple-source-deps - has now been done.

**Comment by johnzhou721** [comment]
@freakboy3742 Is it okay if I reimpl. the Catalyst Support stuff on this PR? A bunch of changes are needed for the logic and I think it'll be easier for me to start a separate PR to here.

**Comment by freakboy3742** [comment]
@johnzhou721 I'm confused about what you're asking. This isn't a PR, and there isn't an open PR for Catalyst support...

**Comment by johnzhou721** [comment]
Oh I was wondering if (when) I do the building stuff for mac catalyst if I could not base my code on https://github.com/beeware/Python-Apple-support/pull/256 and start a separate PR.

**Comment by freakboy3742** [comment]
Use #256 if it's helpful; if it's not, don't. I don't mind either way.
